atomic_reactor/plugins/pre_add_filesystem.py

```diff
@@ -374,6 +329,39 @@ def update_repos_from_composes(self):
             self.log.info('adding repo file from compose: %s', compose_info['result_repofile'])
             self.repos.append(compose_info['result_repofile'])
 
+    def _add_filesystem_to_dockerfile(self, file_name, build_dir: BuildDir):
+        """
+        Put an ADD instruction into the Dockerfile (to include the filesystem
+        into the container image to be built)
+        """
+        content = 'ADD {0} /\n'.format(file_name)
+        lines = build_dir.dockerfile.lines
+
+        for index, line in enumerate(lines):
+            if 'koji/image-build' in line:
+                lines.insert(index+1, content)
```

Comment:
I'm still not entirely convinced by this. I could imagine someone putting a comment about "koji/image-build" in their Dockerfile, and then we would end up putting the ADD instruction after that comment.

```Dockerfile
# OSBS will replace koji/image-build with a filesystem tarball downloaded from Brew
FROM koji/image-build
```
Does [DockerfileParser](https://github.com/containerbuildsystem/dockerfile-parse/blob/master/dockerfile_parse/parser.py) provide anything that could be used to do this more reliably? Maybe the `structure` property?

Side note: inserting to a list while enumerating that list actually works as intended? :open_mouth: (it does, but only if you insert *after* the current index)

Reply:
Note the side-effect that the _inserted_ line will still be checked by the `if`... probably not dangerous here, unless the filesystem archive name contains 'koji/image-build' :grin:

Reply:
updated the logic to read from `structure` and insert into `lines` and use `parent_images` to replace `koji/image-build` with `scratch`